packages/dhis2/src/Adaptor.js

```diff
@@ -634,24 +605,54 @@ export function upsert(
       // NOTE: that these parameters are all expanded by the `get`, `create`, and
       // `update` functions used inside this composed "upsert" function.
       console.log(`Preparing upsert via 'get' then 'create' OR 'update'...`);
-      promise = get(
-        resourceType,
-        query,
-        options
-      )(state).then(resp => {
-        const resources = resp.data[resourceType];
+      promise = request(configuration, {
+        method: 'GET',
+        path: generateUrlPath(
+          configuration,
+          resolvedOptions,
+          resolvedResourceType
+        ),
+        options: {
+          ...resolvedOptions,
+          query: {
+            ...resolvedQuery,
+          },
+        },
+      }).then(resp => {
+        const resources = resp.body[resourceType];
         if (resources.length > 1) {
           throw new RangeError(
             `Cannot upsert on Non-unique attribute. The operation found more than one records for your request.`
```

Comment:
This is such a poor error message.

Maybe we can do use `util.throwError()` to provide a simpler, cleaner explanation of the error and provide a fix.

This should probably be spun out into a new issue